pkg/appfile/parser.go

```diff
@@ -406,3 +435,93 @@ func PrepareProcessContext(k8sClient client.Client, wl *Workload, applicationNam
+func GetOutputSecretNames(workloads *Workload) (string, error) {
+	secretName, err := getComponentSetting(process.OutputSecretName, workloads.Params)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprint(secretName), nil
+}
+
+func parseWorkloadK8sSecretSchema(ctx context.Context, c client.Client, namespace string, wl *Workload) ([]process.RequiredSecrets, error) {
+	var requiredSecret []process.RequiredSecrets
+	api, err := utils.GenerateOpenAPISchemaFromDefinition(wl.Name, wl.Template)
+	if err != nil {
+		if !errors.Is(err, errors.Errorf(utils.ErrNoSectionParameterInCue, wl.Name)) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	schema, err := utils.ConvertOpenAPISchema2SwaggerObject(api)
+	if err != nil {
+		return nil, err
+	}
+	tag := utils.K8sSecretSchemaTag
+	for k, v := range schema.Properties {
+		description := v.Value.Description
+		if strings.Contains(description, tag) {
+			contextName := strings.Split(description, tag)[1]
+			contextName = strings.TrimSpace(contextName)
+			secretNameInterface, err := getComponentSetting(k, wl.Params)
+			if err != nil {
+				return nil, err
+			}
+			secretName := fmt.Sprint(secretNameInterface)
+			secretData, err := extractSecret(ctx, c, namespace, fmt.Sprint(secretName))
+			if err != nil {
+				return nil, err
+			}
+
+			requiredSecret = append(requiredSecret, process.RequiredSecrets{
+				Name:        secretName,
+				ContextName: contextName,
+				Namespace:   namespace,
+				Data:        secretData,
+			})
+		}
+	}
+	return requiredSecret, nil
+}
+
+func extractSecret(ctx context.Context, c client.Client, namespace, name string) (map[string]interface{}, error) {
+	secretData := make(map[string]interface{})
+	var secret v1.Secret
+	if err := c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &secret); err != nil {
+		return nil, fmt.Errorf("failed to get secret %s from namespace %s which is required by the component: %w",
+			name, namespace, err)
+	}
+	for k, v := range secret.Data {
+		secretData[k] = string(v)
+	}
+	if len(secretData) == 0 {
+		return nil, fmt.Errorf("data in secret %s from namespace %s isn't available", name, namespace)
+	}
+	return secretData, nil
+}
+
+func getComponentSetting(settingParamName string, params map[string]interface{}) (interface{}, error) {
+	if secretName, ok := params[settingParamName]; ok {
+		return secretName, nil
+	}
+	return nil, fmt.Errorf("failed to get the value of component setting %s", settingParamName)
+}
+
+// IsCloudResourceProducer checks whether a workload is cloud resource producer role
+func (wl *Workload) IsCloudResourceProducer() bool {
+	var flag bool
+	_, flag = wl.Params[process.OutputSecretName]
+	return flag
+}
+
+// IsCloudResourceConsumer checks whether a workload is cloud resource consumer role
+func (wl *Workload) IsCloudResourceConsumer() bool {
+	matched, err := regexp.Match(`k8sSecretSchema=`, []byte(wl.Template))
```

Comment:
we can use a const for `k8sSecretSchema`

Reply:
I did by using the const https://github.com/oam-dev/kubevela/pull/1264/files#diff-84088786897ac51a95b85eeab65171013b54c6d4b53edf295fb457de4e62becdR50.

But `+` in `+k8sSecretSchema` is special character. I think I should look for a way to escape the character. Thanks.

Reply:
Fixed it by `regexp.QuoteMeta(utils.K8sSecretSchemaTag)`. Thanks for your reminder.